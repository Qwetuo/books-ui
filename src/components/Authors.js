import React, { Component } from "react";

const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";

class Authors extends Component {
  constructor() {
    super();
    this.state = {
      authors: []
    };
  }

  async getAuthors() {
    const url = `${API_HOST}/authors`;
    const response = await fetch(url);
    if (response.ok) {
      const data = await response.json();
      this.setState({
        authors: data
      });
    }
  }

  async componentDidMount() {
    await this.getAuthors();
  }

  render() {
    return (
      <div>
        <h1>AUTHORS</h1>
        {this.state.authors.map(author => {
          return <li key={author._id}>{author.name}</li>;
        })}
      </div>
    );
  }
}

export default Authors;
