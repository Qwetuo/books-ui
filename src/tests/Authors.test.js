import React from 'react';
import Authors from '../components/Authors';

import ShallowRenderer from 'react-test-renderer/shallow'; // ES6

const renderer = new ShallowRenderer();
renderer.render(<Authors />);

it('renders title', () => {
    let result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot()
})

it('renders title and authors after the component mount', async () => {
    fetch.mockResponseOnce(
        JSON.stringify([
            {
                name: "author1",
                age: 1,
                _id: "100"
            },
            {
                name: "author 2",
                age: 2,
                _id: "2"
            }
        ])
    )
    const instance = renderer.getMountedInstance();
    await instance.componentDidMount();
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot()
})