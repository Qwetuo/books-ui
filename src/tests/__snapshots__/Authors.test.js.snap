// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title 1`] = `
<div>
  <h1>
    AUTHORS
  </h1>
</div>
`;

exports[`renders title and authors after the component mount 1`] = `
<div>
  <h1>
    AUTHORS
  </h1>
  <li>
    author1
  </li>
  <li>
    author 2
  </li>
</div>
`;
